docs/user-guide/connecting-applications.md

```diff
@@ -0,0 +1,205 @@
+
+## Exposing nginx pods to the cluster
+
+We did this in a previous example, but lets do it once again and focus on the networking perspective. Create an nginx pod, and note that it has a container port specification:
+```yaml
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: my-nginx
+spec:
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx
+        ports:
+        - containerPort: 80
+```
+
+This makes it accessible from any node in your cluster. Check the nodes the pod is running on:
+```shell
+$ kubectl get pods -l app=nginx -o wide
+my-nginx-6isf4   1/1       Running   0          2h        e2e-test-beeps-minion-93ly
+my-nginx-t26zt   1/1       Running   0          2h        e2e-test-beeps-minion-93ly
+```
+
+Check your pods ips:
+```shell
+$ kubectl get pods -l app=nginx -o json | grep podIP
+                "podIP": "10.245.0.15",
+                "podIP": "10.245.0.14",
+```
+You should be able to ssh into any node in your cluster and curl both ips. Note that the containers are *not* using port 80 on the node, nor are there any special NAT rules to route traffic to the pod. This means you can run multiple nginx pods on the same node all using the same containerPort and access them from any other pod or node in your cluster using ip. Like Docker, ports can still be published to the host node's interface(s), but the need for this is radically diminished because of the networking model.
+
+You can read more about [how we achieve this](../networking.md#how-to-achieve-this) if you’re curious.
+
+## Creating a Service for the pods
+
+So we have pods running nginx in a flat, cluster wide, address space. In theory, you could talk to these pods directly, but what happens when a node dies? The pods die with it, and the replication controller will create new ones, with different ips. This is the problem a Service solves.
+
+A Kubernetes Service is an abstraction which defines a logical set of Pods running somewhere in your cluster, that all provide the same functionality. When created, each Service is assigned a unique IP address (also called clusterIP). This address is tied to the lifespan of the Service, and will not change while the Service is alive. Pods can be configured to talk to the Service, and know that communication to the Service will be automatically load-balanced out to some pod that is a member of the Service ([why not use round robin dns?](../services.md#why-not-use-round-robin-dns)).
```

Comment:
I planned for that to go into a doc about "advanced use cases", along with setting up your own load balancer, importing services (nil selector), session affinity, and some non-service-related topics.